File: 26-remove-duplicates-from-sorted-array/remove-duplicates-from-sorted-array.py
class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:
        # sorted array duplicates - means they should be adjacent to each other 

        # create a new start index k - for non-repeating elememts
        if len(nums) <=1:
            return len(nums)
        k = 0
        i = 0
        for i in range(len(nums)-1):
            if nums[i] != nums[i+1]:
                nums[k] = nums[i]
                k+=1
        nums[k] = nums[i+1]
        
        return k+1

         

File: 26-remove-duplicates-from-sorted-array/remove-duplicates-from-sorted-array.cpp
class Solution {
public:
    int removeDuplicates(vector<int>& nums) {
        // sorted array duplicates - means they should be adjacent to each other 

        // create a new start index k - for non-repeating elememts
        int k = 0;
        for (int i=0; i < nums.size()-1; i++){
            if (nums[i] != nums[i+1]){
                nums[k] = nums[i];
                k+=1;
            }
        nums[k] = nums[i+1];
        }
        return k+1;
    }
};